Fix problem with `libsqlite 3.49.1`

After the CI upgraded to `libsqlite 3.49.1` we get the errors 
```
FAILED tests/test_cache.py::TestCache::test_delete - sqlite3.OperationalError: no such column: "size" - should this be a string literal in single-quotes?
```
https://github.com/fau-advanced-separations/CADET-Process/actions/runs/13410222197/job/37458512256#step:8:704

This has been tested by me locally to be caused by `libsqlite 3.49.1` and can be solved by downgrading to `libsqlite 3.48.0`, but pinning that version isn't a good permanent fix.

--- a/environment.yml
+++ b/environment.yml
@@ -4,4 +4,5 @@ channels:
 dependencies:
   - python>=3.10.*
   - cadet>=5.0.3
+  - libsqlite<3.49
   - pip

